### DISCUSSION TITLE
Note about ESLint "no-undef" rule

### DISCUSSION THREAD

**Discussion by GekkotaDev** [vue-macros/vue-macros#768] [General]
Something that may confuse people would be the built-in ESLint rule `no-undef` stating the macros are undefined despite Typescript having no complaints and the macros behaving correctly at runtime.

![image](https://github.com/user-attachments/assets/32a2c11e-1e18-47bf-859d-38236acbd925)
![image](https://github.com/user-attachments/assets/560f1292-b96d-405b-ba41-906bd3f7f9fa)

Not an issue of Vue Macros itself but nevertheless suggesting this behavior from ESLint should be noted in documentation to reduce confusion, and a simple fix for users would be to add the following lines

```js
// eslint.config.js

/*
  For Nuxt 3 users, assume that this config is wrapped with `withNuxt` from
  "./.nuxt/eslint.config.mjs"
 */
export default [
    {
        languageOptions: {
            globals: {
                $: "readonly",
                $$: "readonly",
                $computed: "readonly",
                $customRef: "readonly",
                $defineModels: "readonly",
                $defineProps: "readonly",
                $ref: "readonly",
                $shallowRef: "readonly",
                $toRef: "readonly",
                defineModels: "readonly",
                definePropsRefs: "readonly",
                defineRender: "readonly",
            }
        }
    }
];
```

Alternatively, the rule `no-undef` may be disabled completely

```js
// eslint.config.js

export default [
    {
        rules: {
            "no-undef": 0,
        }
    }
];
```

**Comment by sxzz** [comment]
We have `@vue-macros/eslint-config` package that you can use, even though it's not mentioned in the documentation.

**Comment by sxzz** [comment]
See https://vue-macros.dev/guide/eslint-integration.html
